src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -360,6 +342,29 @@ private void methodEpilogue() {
         mv.visitEnd();
     }
 
+    /**
+     * Returns an object to pass this.classData to the <clinit> method of the
```

Comment:
What about:

```suggestion
     * Returns the class data object that will be passed to `Lookup.defineHiddenClass`.
     * The classData is loaded in the <clinit> method of the generated class.
     * If the class data contains only one single object, this method returns  that single object.
     * If the class data contains more than one objects, this method returns a List.
     *
     * This method returns null if no class data.
```

Reply:
Actually, the classData is passed here:

```
    private MemberName loadMethod(byte[] classFile) {
        Class<?> invokerClass = LOOKUP.makeHiddenClassDefiner(className, classFile, Set.of())
                                      .defineClass(true, classDataValues());
        return resolveInvokerMember(invokerClass, invokerName, invokerType);
    }
```

So it doesn't go through `Lookup.defineHiddenClass`.

Reply:
yes, it's an internal version that defines a hidden class, equivalent to calling `Lookup::defineHiddenClassWithClassData` (typo in my suggested wording), but it will bypass the security permission check and also returns the Class instead of Lookup (saving not to allocate a Lookup object).